Pin and Separate Days in Week at a Glance

jrjohnson opened the pull request:
It can be difficult in a busy week to see the split between days, and in a busy day as you scroll to remember where you are on the page. Adding a border to split the days makes it easier and as you scroll down the page the day of the week pins itself to the top. This allows us to remove the day from each even making them easier to read as well.

dartajax commented:
<img width="2850" height="1146" alt="image" src="https://github.com/user-attachments/assets/3f105539-20de-4c19-86c0-cbfe8f0cb5df" />

jrjohnson commented:
Cleaned that up @dartajax and fixed the overrun when the screen is smaller by replacing the full day name with shorter versions. Opening it up for code review, but keep any feedback coming!

dartajax commented:
I would be glad to go with it the way it is but is it possible to line these ups all the way vertically?

dartajax commented:
> Sure @dartajax, but I'm not sure what way to line it up. Can you mess with a screenshot and a line and let me know what should line up with what? Top of Day top of event? Or maybe just remove the space over the first event so it's closer to the orange line?

That would work - I'd suggest removing the space over the first event so it ...

a. goes closer to the orange line and 
b. matches the top of the day label too

or c. settle on a compromise version of vertical spacing they can share 

my screen shot is not the hottest but here you go ...

<img width="2278" height="510" alt="image" src="https://github.com/user-attachments/assets/78e406a2-efcf-4aaa-8721-77065b9af3fe" />
